include/kernel/thread.h

```diff
@@ -307,5 +307,7 @@ void awaken(void);
 /* Place the thread bound to this scheduling context in the release queue
  * of periodic threads waiting for budget recharge */
 void postpone(sched_context_t *sc);
+#else
+static inline void updateTimestamp(void) {}
```

Comment:
Given we need the function in MSC and non-MCS now (even if it is just a dummy for non-MSC to avoid to much conditional parts), I suggest to move it out of the condition block now and instead create a conditional block for the function body. Having just one definition in the code makes it easier to find the actual implementation.
The compiler should be able to optimize this out, but it may have an impact on verification. In the declined PR https://github.com/seL4/seL4/pull/752 I added a macro `MCS_TERNARY()` to wrap this and avoid impacting verification.

Reply:
Agreed and done.